src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -26,9 +26,9 @@ internal sealed class CoreClrAssemblyLoader
         private static string _msbuildDirPath;
         private static readonly Version _currentAssemblyVersion = new Version(Microsoft.Build.Shared.MSBuildConstants.CurrentAssemblyVersion);
 
-        internal CoreClrAssemblyLoader()
+        static CoreClrAssemblyLoader()
         {
-            _msbuildDirPath = FileUtilities.NormalizePath(BuildEnvironmentHelper.Instance.CurrentMSBuildExePath);
+            _msbuildDirPath = FileUtilities.NormalizePath(typeof(CoreClrAssemblyLoader).Assembly.Location);
```

Comment:
ComparePathsNoThrow handles the normalization part of this for you.

Reply:
Thanks @forgind.  I'd like to leave this here.  The reason is that it happens only once per process, and has the ability to simplify the checks that must happen per assembly, instead of just once per process.  If we don't do this here, and the value would have changed, then when loading an assembly, we'd have to do a string comparison which would fail, and then we'd have to normalize both inputs, and then try again.

If on the other hand, the initial normalize call here changed the input, it could possibly avoid the extra normalizations per assembly load.